### PR TITLE
Run tests and PHPStan on PHP 8.5

### PR DESCRIPTION
**Pull request by kukulich**
Psalm is now broken because of this: https://github.com/php/php-src/pull/19424

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -110,6 +112,8 @@ jobs:
           - "locked"
         php-version:
           - "8.4"
+          # Psalm currently fails
+          # - "8.5"
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml:116]: I suppose MT will also need to wait for an upgrade

File: test/unit/Fixture/PHP85ClassForSourceStubber.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+{
+    public function methodFromTraitToAlias()
+    {
+    }
+}
+
+/**
+ * Class comment
+ */
+abstract class PHP85ClassForSourceStubber extends ParentClassForSourceStubber implements ImplementedInterfaceForSourceStubber, Serializable
```
Comment (Ocramius) [inline test/unit/Fixture/PHP85ClassForSourceStubber.php]: This class is filled to the brim with small edge cases: are there specific sections from PHP 8.5 that were relevant, or is this an "upgrade of the stubbing with some additions"?

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 3]
```diff
@@ -133,6 +133,16 @@ static function (string $className): bool {
                         return false;
                     }
 
+                    // Missing in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Can we reference an upstream issue, please?

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 4]
```diff
@@ -188,8 +198,17 @@ private function assertSameClassAttributes(CoreReflectionClass $original, Reflec
         $this->assertSameInterfaces($original, $stubbed);
 
         foreach ($original->getMethods() as $method) {
+            $methodName = $original->getName() . '#' . $method->getName();
+
+            // Needs fixes in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Can we reference an upstream issue, please?

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 5]
```diff
@@ -204,10 +223,19 @@ private function assertSameClassAttributes(CoreReflectionClass $original, Reflec
             $originalConstantName = $originalConstant->getName();
             assert($originalConstantName !== '');
 
+            $constantName = $original->getName() . '::' . $originalConstant->getName();
+
+            // Needs fixes in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Can we reference an upstream issue, please?

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 6]
```diff
@@ -301,6 +329,20 @@ public static function internalFunctionsProvider(): array
                 static function (string $functionName): bool {
                     $reflection = new CoreReflectionFunction($functionName);
 
+                    // Missing in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Can we reference an upstream issue, please?

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 7]
```diff
@@ -361,6 +403,18 @@ public static function internalConstantsProvider(): array
             }
 
             foreach ($extensionConstants as $constantName => $constantValue) {
+                // Missing in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Can we reference an upstream issue, please?

**Review by Ocramius — APPROVED** [review]
LGTM!

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php (hunk 3 above)
Comment (kukulich) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: I have to admit I'm lazy (and also time limited) to create issue/PR. I hope they will fix with another release in December as always. :)

Comment (Ocramius) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: I'd be happy to have a single placeholder for all of these, if possible: after all, you did the discovery work yourself, and it's a matter of reporting on it.

Alternatively, I can do this later today, if you want

File: test/unit/Fixture/PHP85ClassForSourceStubber.php (hunk 2 above)
Comment (kukulich) [reply test/unit/Fixture/PHP85ClassForSourceStubber.php]: You are right, I've just forcepushed a small version with just the one change in PHP 8.5.

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php (hunk 3 above)
Comment (kukulich) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: > Alternatively, I can do this later today, if you want

👍 Thank you.

Comment (Ocramius) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: @kukulich I raised https://github.com/JetBrains/phpstorm-stubs/pull/1781 with the full list: can you update the various TODOs?

I don't care if the upstream PR ("issue") is closed, but at least I centralized this knowledge somewhere

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php (hunk 4 above)
Comment (kukulich) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php]: Done.

**Review by Ocramius — APPROVED** [review]
:ship:

**Comment by kukulich** [comment]
@Ocramius Btw it looks renovate is broken again.

**Comment by Ocramius** [comment]
@kukulich any details on what broke?

**Comment by kukulich** [comment]
I think Psalm is not updated: https://github.com/Roave/BetterReflection/blob/6.61.x/tools/composer.json

https://github.com/vimeo/psalm/releases/tag/6.13.1

**Comment by Ocramius** [comment]
@kukulich I think it's because of `roave/infection-static-analysis-plugin` locking it in place. I haven't touched `roave/infection-static-analysis-plugin` in a while: that should allow an upgrade again.

Renovate will always struggle to upgrade them separately

**Comment by kukulich** [comment]
Oh, ok 👍

**Comment by Ocramius** [comment]
I released `1.39.0` of `roave/infection-static-analysis-plugin`, meanwhile: `infection/infection` has been rolled back for now, and will be upgraded separately in that package
